pkg/action/archive_test.go

```diff
@@ -237,14 +238,21 @@ func TestScanArchive(t *testing.T) {
 	}
 
 	outBytes := out.Bytes()
+	sort.Slice(outBytes, func(i, j int) bool {
```

Comment:
If I understand this: this is sorting the entire byte stream one character at a time.

If we're looking for help with non-determinism, I feel like we should convert the byte stream to a slice of strings and sort that instead. To be honest though, I think comparing the JSON output would be the correct answer for this test, rather than simple. 

Since this is an archive test, we really only need the list of files found in the JSON.

Reply:
I was thinking the same thing. I'll make that change.

Reply:
This will require some additional work in how we display the paths of files scanned within archives. Currently they have the entire temporary path which will never be the same across test runs so that needs to be removed when the reports are built.

Reply:
Updated in [`f2d2599` (#617)](https://github.com/chainguard-dev/malcontent/pull/617/commits/f2d2599b9505783c903b658b6b396c1c9b19b786).